src/OrchardCore/OrchardCore.Abstractions/BackgroundTasks/BackgroundTaskContext.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+using OrchardCore.Environment.Shell.Scope;
+
+namespace OrchardCore.BackgroundTasks
+{
+    public class BackgroundTaskContext
+    {
+        public BackgroundTaskContext(string name, ShellScope scope)
+        {
+            Name = name;
+            Tenant = scope.ShellContext.Settings.Name;
+            Services = scope.ServiceProvider;
+        }
+
+        public string Name { get; }
+        public string Tenant { get; }
+        public IServiceProvider Services { get; }
```

Comment:
What can this be used for? `IBackgroundTaskEventHandler` is already resolved from the same scope so any implementation will have access to the same services.

Reply:
Yes, as you want, just because this is what we do for the the `IBackgroundTask.DoWorkAsync()` where we pass the scoped service provider. So, yes i already thought about this, and also thought about removing the service provider param passed to `DoWorkAsync()`, letting scoped `IBackgroundTask` injecting what they need.

But would be a breaking change so i let the `DoWorkAsync()` signature as it was done originally (not by me), hmm maybe at some point, before the hosted bg service, the `IBackgroundTask` was resolved as singletons and then `DoWorkAsync()` called in a scoped context.

So, if for some reasons you want to register an `IBackgroundTask` as a singleton and that get executed in a scope context, passing a service provider still allows to resolve scoped services that you would not be able to inject through a singleton ctor. That's why i did the same, but through a context, for `IBackgroundTaskEventHandler`.

Sorry for the long story, anyway i will remove this service provider from the background context ;)

Reply:
I'm not sure I'm following :). In my understanding, the scope of the event handler is not tied to how the task is registered. Or do you mean that you may register `IBackgroundTaskEventHandler` as a singleton, thus it wouldn't necessarily resolved from the current `scope`? I see, that's interesting. In that case, I see why this may be necessary here and thus the property is useful.

Reply:
> I'm not sure I'm following :). In my understanding, the scope of the event handler is not tied to how the task is registered.

Yes, you're right

> Or do you mean that you may register IBackgroundTaskEventHandler as a singleton, thus it wouldn't necessarily resolved from the current scope? I see, that's interesting. In that case, I see why this may be necessary here and thus the property is useful.

Yes, that's what i was thinking about, normally this is not a recommanded pattern, but in some cases could be useful

Reply:
I see now and I agree, thanks for the explanation. Then I have no objection against this property.